src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/InfraProvisioningWizard.tsx

```
@@ -260,106 +316,204 @@ export const InfraProvisioningWizard: React.FC<InfraProvisioningWizardProps> = p
+        const v1YAMLAsJSON: Record<string, any> =
+          configuredOption &&
+          StarterConfigIdToOptionMap[configuredOption.id] === PipelineConfigurationOption.GenerateYAML
+            ? generatedYAMLAsJSON
+            : getCIStarterPipelineV1()
+        return createPipelineV2Promise({
+          body: CI_YAML_VERSIONING
+            ? yamlStringify(
+                storeInGit
+                  ? v1YAMLAsJSON
+                  : addRepositoryInfoToPipeline({
+                      currentPipeline: v1YAMLAsJSON,
+                      connectorRef,
+                      repoName:
+                        (selectRepositoryRef.current?.repository?.namespace &&
+                        existingGitConnectorUrl.endsWith(selectRepositoryRef.current.repository.namespace)
+                          ? selectRepositoryRef.current?.repository?.name
+                          : getFullRepoName(selectRepositoryRef.current?.repository)) || ''
+                    })
+              )
+            : constructPipelinePayloadWithCodebase(selectRepositoryRef.current.repository),
           queryParams: {
             accountIdentifier: accountId,
             orgIdentifier,
-            projectIdentifier
+            projectIdentifier,
+            ...(shouldSavePipelineToGit && {
+              ...gitParams,
+              filePath: yamlPath,
+              commitMsg: `${getString('common.addedEntityLabel', {
+                entity: getString('common.pipeline').toLowerCase()
+              })} ${yamlPath}`,
+              ...(isGitSaveRetry && { isNewBranch: isGitSaveRetry, baseBranch: defaultBranch })
+            })
           },
           requestOptions: { headers: { 'Content-Type': 'application/yaml' } }
+        }).then((createPipelineResponse: ResponsePipelineSaveResponse) => {
+          const { status } = createPipelineResponse
+          if (status === Status.SUCCESS && createPipelineResponse?.data?.identifier) {
```

I feel like 377 to 442 could be cleaned up a bit if we switch to async await - it could make the nested .then() easier to understand

also we could probably have a single catch instead of line 430 and line 436 :)

---

The individual catches on lines 430 and 436 are for their respective promises, if the inner promise errors out, unless we do not re-throw the error from it, `catch` of the outer promise won't catch it right? To avoid this, I gave inner promise it's own `catch` block. 

Is there a better way about this?

---

the two catches are doing the same thing - so I think one would suffice - basically the nested promise (which would now be an await) should throw an error and that would be caught by the single catch - as it is all inside a single try

---

Done, cleaned it up, thanks for this great feedback!